prow/oss/terraform/modules/alerts/boskos/main.tf

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_monitoring_alert_policy" "boskos-alerts" {
+  count        = var.enabled ? 1 : 0
+  project      = var.project
+  display_name = "boskos-alerts"
+  combiner     = "OR" # required
+
+  conditions {
+    display_name = "Boskos ran out of resources"
+
+    condition_monitoring_query_language {
+      duration = "0s"
+      query    = <<-EOT
+      fetch k8s_container
+      | metric 'workload.googleapis.com/boskos_resources'
+      | {
+          t_0:
+          filter state == 'free'
+          ;
+          t_1:
+          ident
+      }
+      | group_by [metric.type, metric.instance]
```

Comment:
I think `sum` is the default aggregator, but it might be clearer to explicitly `sum(val())`.

Reply:
I have tried `sum(val())` in UI, it doesn't work, `sum` worked but aggregated values from all types, which is not what we want